src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt

```diff
@@ -83,20 +83,21 @@ class ValidationException(val errorCode: ErrorCode) : RuntimeException()
 class ErrorResponse(val status: Int, errorCode: ErrorCode) {
   constructor(status: HttpStatus, errorCode: ErrorCode) : this(status.value(), errorCode)
 
-  val errorCode = errorCode.name
+  val errorCode = errorCode.code
   val userMessage = errorCode.userMessage
 }
 
-enum class ErrorCode(val userMessage: String) {
-  // Standard errors
-  AUTH("Authentication failure"),
-  DOWNSTREAM("An error occurred calling a downstream service"),
-  INTERNAL_ERROR("An unexpected error occurred"),
-  MALFORMED_REQUEST("Failed to read the payload"),
-  NOT_FOUND("Not found"),
-  // Custom errors
-  EMAIL_MANDATORY("The email address must be entered"),
-  EMAIL_TOO_LONG("The email address can have a maximum length of $MAX_EMAIL_LENGTH"),
-  INVALID_EMAIL("Enter an email address in the correct format"),
-  INVALID_CJSM_EMAIL("Enter an email address which ends with 'cjsm.net'"),
-}
+sealed class ErrorCode(val code: String, val userMessage: String)
+
+sealed class StandardErrorCodes(code: String, userMessage: String) : ErrorCode(code, userMessage)
+object AuthenticationError : StandardErrorCodes("AUTH", "Authentication failure")
+object DownstreamError : StandardErrorCodes("DOWNSTREAM", "An error occurred calling a downstream service")
+object InternalError : StandardErrorCodes("INTERNAL_ERROR", "An unexpected error occurred")
+object MalformedRequest : StandardErrorCodes("MALFORMED_REQUEST", "Failed to read the payload")
+object NotFound : StandardErrorCodes("NOT_FOUND", "Not found")
+
+sealed class MagicLinkRequestErrorCodes(code: String, userMessage: String) : StandardErrorCodes(code, userMessage)
+object EmailMandatory : MagicLinkRequestErrorCodes("EMAIL_MANDATORY", "The email address must be entered")
+object EmailTooLong : MagicLinkRequestErrorCodes("EMAIL_TOO_LONG", "The email address can have a maximum length of $MAX_EMAIL_LENGTH")
+object EmailInvalid : MagicLinkRequestErrorCodes("INVALID_EMAIL", "Enter an email address in the correct format")
+object EmailInvalidCjsm : MagicLinkRequestErrorCodes("INVALID_CJSM_EMAIL", "Enter an email address which ends with 'cjsm.net'")
```

Comment:
The main change of the PR - allows us to extend the set of standard errors with custom errors for each endpoint.

Reply:
Like this 👍🏼